dvc/external_repo.py

```diff
@@ -234,17 +239,16 @@ def _clone_default_branch(url, rev, for_write=False):
     return clone_path, shallow
 
 
-def _unshallow(git):
-    if git.gitpython.repo.head.is_detached:
-        # If this is a detached head (i.e. we shallow cloned a tag) switch to
-        # the default branch
-        origin_refs = git.gitpython.repo.remotes["origin"].refs
-        ref = origin_refs["HEAD"].reference
-        branch_name = ref.name.split("/")[-1]
-        branch = git.gitpython.repo.create_head(branch_name, ref)
-        branch.set_tracking_branch(ref)
-        branch.checkout()
-    git.pull(unshallow=True)
+def _merge_upstream(git):
+    from scmrepo.exceptions import SCMError
+
+    try:
+        branch = git.active_branch()
+        upstream = f"refs/remotes/origin/{branch}"
```

Comment:
Is it okay to hardcode this? IIRC the default remote name is configurable in git.

Reply:
dulwich clone doesn't actually check `clone.defaultRemoteName` (it only supports overriding it directly with `origin=...` equivalent to `git clone --origin <name>`), so for our purposes it will always be `origin`.

I think this is ok for now, and ocne https://github.com/iterative/scmrepo/issues/41 is addressed (and erepo is moved into scmrepo) we should add the methods needed to check git config vars, and update our dulwich clone wrapper to use `clone.defaultRemoteName` if the user actually has it set.